packages/core/src/components/forms/numericStepper.tsx

```diff
@@ -0,0 +1,255 @@
+
+    /** Ref handler that receives HTML `<input>` element backing this component. */
+    inputRef?: (ref: HTMLInputElement) => any;
+
+    /** Name of icon (the part after `pt-icon-`) to render on left side of input. */
+    leftIconName?: string;
+
+    /** Placeholder text in the absence of any value. */
+    placeholder?: string;
+
+    /**
+     * Increment between successive values when "Shift" is held.
+     * @default 10
+     */
+    majorStepSize?: number;
+
+    /** Maximum value of the input. */
+    max?: number;
+
+    /** Minimum value of the input. */
+    min?: number;
+
+    /**
+     * Increment between successive values when "Alt" is held.
+     * @default 0.1
+     */
+    minorStepSize?: number;
+
+    /**
+     * Increment between successive values when no modifier keys are held.
+     * @default 1
+     */
+    stepSize?: number;
+
+    /**
+     * Value to display in the input field
+     * @default ""
+     */
+    value?: string;
+
+}
+
+export interface INumericStepperState {
+    shouldSelectAfterUpdate: boolean;
+    value: string;
+}
+
+@PureRender
+export class NumericStepper extends React.Component<HTMLInputProps & INumericStepperProps, INumericStepperState> {
+    public static displayName = "Blueprint.NumericStepper";
+
+    public static defaultProps: INumericStepperProps = {
+        buttonPosition: "right",
+        majorStepSize: 10,
+        minorStepSize: 0.1,
+        stepSize: 1,
+        value: "",
+    };
+
+    private static DECREMENT_KEY = "decrement";
+    private static INCREMENT_KEY = "increment";
+
+    private static DECREMENT_ICON_NAME = "minus";
+    private static INCREMENT_ICON_NAME = "plus";
+
+    private input: HTMLInputElement;
+
+    public constructor(props?: HTMLInputProps & INumericStepperProps) {
+        super(props);
+        this.state = {
+            shouldSelectAfterUpdate: false,
+            value: props.value,
+        };
+    }
+
+    public render() {
+        const { buttonPosition, className } = this.props;
+
+        const inputGroup = this.renderInputGroup();
+        const decrementButton = this.renderButton(
+            NumericStepper.DECREMENT_KEY, NumericStepper.DECREMENT_ICON_NAME, this.handleDecrementButtonClick);
+        const incrementButton = this.renderButton(
+            NumericStepper.INCREMENT_KEY, NumericStepper.INCREMENT_ICON_NAME, this.handleIncrementButtonClick);
+
+        let elems: JSX.Element[];
+
+        if (buttonPosition === "left") {
+            elems = [decrementButton, incrementButton, inputGroup];
+        } else if (buttonPosition === "split") {
+            elems = [decrementButton, inputGroup, incrementButton];
+        } else if (buttonPosition === "right") {
+            elems = [inputGroup, decrementButton, incrementButton];
+        } else {
+            elems = [inputGroup];
+        }
+
+        const classes = classNames(Classes.CONTROL_GROUP, { [Classes.DISABLED]: this.props.disabled }, className);
```

Comment:
disabled means nothing on a control group--it only has meaning on the children. so you can skip this and just pass the prop to the interactive children.

> Note that `.pt-control-group` does not cascade any modifiers to its children.
http://blueprintjs.com/docs/#components.forms.control-group